pipelinewise/fastsync/commons/split_gzip.py

```diff
@@ -0,0 +1,127 @@
+                        (Default 0.12)
+
+    Return:
+        File like object
+    """
+    if mode not in ['wb', 'wt']:
+        raise ValueError('Invalid mode: %r' % (mode,))
+    return SplitGzipFile(base_filename, mode, chunk_size_mb, max_chunks, est_compr_rate)
+
+
+# pylint: disable=R0902
+class SplitGzipFile(io.BufferedIOBase):
+    """The SplitGzipFile file like object class that implements only the write method.
+
+    This class only supports writing files in binary mode.
+    """
+    def __init__(self,
+                 base_filename,
+                 mode: str = None,
+                 chunk_size_mb: int = None,
+                 max_chunks: int = None,
+                 est_compr_rate: float = None):
+        super().__init__()
+
+        self.base_filename = base_filename
+        self.mode = mode
+        self.chunk_size_mb = chunk_size_mb if chunk_size_mb is not None else DEFAULT_CHUNK_SIZE_MB
+        self.max_chunks = max_chunks if max_chunks is not None else DEFAULT_MAX_CHUNKS
+        self.est_compr_rate = est_compr_rate if est_compr_rate is not None else EST_COMPR_RATE
+
+        self.split_large_files = True
+        self.chunk_seq = 1
+        self.current_chunk_size_mb = 0
+        self.chunk_filename = None
+        self.chunk_file = None
+
+    def _gen_chunk_filename(self) -> str:
+        """
+        Generates a chunk filename
+
+        Pattern if max_chunks is zero: <base_filename>
+        Pattern if max_chunks is greater than zero: <base_filename>.part<chunk-number-padded-five-digits>
+
+        Returns:
+            string
+        """
+        if self.max_chunks == 0:
+            chunk_filename = self.base_filename
+        else:
+            if self.current_chunk_size_mb >= self.chunk_size_mb and self.chunk_seq < self.max_chunks:
+                # Increase the chunk sequence and reset size to zero
+                self.chunk_seq += 1
+                self.current_chunk_size_mb = 0
+
+            chunk_filename = f'{self.base_filename}.part{self.chunk_seq:05d}'
+
+        return chunk_filename
+
+    def _activate_chunk_file(self):
+        """
+        Activate a file like object to write data into the active chunk
+        """
+        chunk_filename = self._gen_chunk_filename()
+        # Close the actual chunk file if exists and open a new one
+        if self.chunk_filename != chunk_filename:
+            if self.chunk_file:
+                self.chunk_file.close()
+
+            # Open the actual chunk file with gzip data writer
+            self.chunk_filename = chunk_filename
+            self.chunk_file = gzip.open(self.chunk_filename, self.mode)
+
+    @staticmethod
+    def _bytes_to_megabytes(size: int) -> float:
+        """
+        Transforms bytes ot megabytes
+
+        Args:
+            size: Number of bytes
+        Returns:
+            floating point number number
+        """
+        return size / float(1 << 20)
+
+    def write(self, b):
```

Comment:
Can you add docstring to this method? also is it possible to rename `b` to a better name? I have no idea what is.

Reply:
added by https://github.com/transferwise/pipelinewise/pull/696/commits/b31bc15ea30f29f394a1d2a1d6b4d3309482a92c